pylearn2/models/mlp.py

```diff
@@ -3237,9 +3245,16 @@ def fprop(self, state_below):
                                                        "pooling.")
 
             if self.pool_type == 'max':
-                p = max_pool(bc01=d, pool_shape=self.pool_shape,
-                             pool_stride=self.pool_stride,
-                             image_shape=self.detector_space.shape)
+                if dnn_available():
```

Comment:
I think I would move the if inside max_pool. That way, we are sure that the faster version will be use at all place max_pool is used. (and remove duplicate code)